startup_scripts/looker

```diff
@@ -1,11 +1,25 @@
 #!/bin/sh
 
 cd $HOME/looker 
-# set your java memory- there should be over 1.5G of system memory 
-# left to run the OS
-MEM=`cat /proc/meminfo | grep MemTotal | awk '{print $2}'`
-JM=`expr $MEM \* 6 / 10`
-JAVAMEM="${JM}k"
+# set your java memory- heap set to 60% of max OS or if there's a cgroup limit
+memTotalKb="$(awk -F ':? +' '$1 == "MemTotal" { print $2; exit }' /proc/meminfo)"
+if [ -r /sys/fs/cgroup/memory/memory.limit_in_bytes ]; then
+    # "18446744073709551615" is a valid value for "memory.limit_in_bytes", which is too big for Bash math to handle
+    # "$(( 18446744073709551615 / 1024 ))" = 0; "$(( 18446744073709551615 * 40 / 100 ))" = 0
+    memLimitKb="$(awk -v totKb="$memTotalKb" '{
+        limB = $0;
+        limKb = limB / 1024;
+        if (!totKb || limKb < totKb) {
+            printf "%.0f\n", limKb;
+        }
+    }' /sys/fs/cgroup/memory/memory.limit_in_bytes)"
+else
+    memLimitKb="$memTotalKb"
+fi
+JAVAMEM="$(awk -v limKb="$memLimitKb" 'BEGIN{
+        limKb = limKb * 6 / 10;
```

Comment:
I kept this math the same, but using 60% feels a bit wrong to me, if your goal is to basically be `limKb = limKb - (1024 * 1024 * 1.5);`

So let me know if you want me to accommodate for that instead, I just didn't wanna change behavior.

Reply:
It seems imo, if we have a cgroup, we can set the limit to exactly the cgroup (maybe subtract a few MB to keep some breathing room), and apply the 60% or some -1.5GB of the value comes from the whole system's memory. If we're running within a container, I'm not sure why we'd need to have that much breathing room.

Reply:
[The JVM has support for containers now](https://www.ibm.com/docs/en/sdk-java-technology/8?topic=options-xx-usecontainersupport), so you can just use something like: `-XX:MaxRAMPercentagerather=90`.

Reply:
This is probably irrelevant now since as you mentioned the JVM now supports automatically setting the heap size when it detects it is running in a container.

However, when running on Ubuntu 20.04.5 LTS the `/sys/fs/cgroup/memory/memory.limit_in_bytes` file seems to have been replaced by `/sys/fs/cgroup/memory.max`.

For me, running this in docker, the memory.max file stores the max memory limit in bytes.
```
$ cat /sys/fs/cgroup/memory.max
4294967296
```

Reply:
Excuse my ignorance of cgroups, this seems to be a platform issue, when running Docker Desktop on MacOS `/sys/fs/cgroup/memory.max` is the file but when running on a Linux node the file was `/sys/fs/cgroup/memory/memory.limit_in_bytes`